src/ast.c

```diff
@@ -25,20 +25,68 @@ extern "C" {
 #pragma warning(disable:4335)
 #endif
 
+JL_DEFINE_MUTEX(flisp)
+
 static uint8_t flisp_system_image[] = {
 #include <julia_flisp.boot.inc>
 };
 
-static fltype_t *jvtype=NULL;
+typedef struct _jl_ast_context_list_t {
+    struct _jl_ast_context_list_t *next;
+    struct _jl_ast_context_list_t **prev;
+} jl_ast_context_list_t;
+
+STATIC_INLINE void jl_ast_context_list_insert(jl_ast_context_list_t **head,
+                                              jl_ast_context_list_t *node)
+{
+    jl_ast_context_list_t *next = *head;
+    if (next)
+        next->prev = &node->next;
+    node->next = next;
+    node->prev = head;
+    *head = node;
+}
+
+STATIC_INLINE void jl_ast_context_list_delete(jl_ast_context_list_t *node)
+{
+    if (node->next)
+        node->next->prev = node->prev;
+    *node->prev = node->next;
+}
+
+typedef struct _jl_ast_context_t {
+    fl_global_context_t fl;
+    fltype_t *jvtype;
+
+    value_t true_sym;
+    value_t false_sym;
+    value_t error_sym;
+    value_t null_sym;
```

Comment:
These symbols are constants, and not allocated on flisp's managed heap, so it should be possible to share them among all contexts.